app/controllers/spree/orders_controller.rb

```diff
@@ -22,6 +23,7 @@ class OrdersController < ::BaseController
     prepend_before_action :require_distributor_chosen, only: :edit
     before_action :check_hub_ready_for_checkout, only: :edit
     before_action :check_at_least_one_line_item, only: :update
+    before_action :hide_ofn_navigation, only: [:show, :edit]
```

Comment:
I wonder if this will work..
After the `@order` variable has been set, we check it and pass the distributor on:

```suggestion
    after_action only: [:show, :edit] do
      hide_ofn_navigation(@order.distributor)
    end
```

(TIL you can pass a block to `_action` filters).
Then...
```diff ruby
-def hide_ofn_navigation
+def hide_ofn_navigation(distributor = current_distributor)
```

Reply:
I've tested, but It does not work (unfortunately because it's beautiful). 

As far as I understood, it's because `after_action` is run after the view is _rendered_ (or at least filled and cached) and then we don't initialize `@hide_ofn_navigation` with its right value (it's `nil` then). 

But this gave me an idea, I think we can get that order instance a `before_action` ; that's why I give a try in that commit: a1b6f18886

(but i'ts not as simple, since `show` and `edit` methods does not retrieve `@order` in a same way)

Reply:
> `after_action` is run after the view is _rendered_ (or at least filled and cached)

Ahh, yes that seems rather obvious now! Good idea. 
I noticed that we already have the `set_current_order` method, so have pushed up one last suggestion which I think is a bit tidier.